### PR TITLE
Move csr_accessor from dal::detail to dal:: namespace

### PR DESCRIPTION
**Pull request by Vika-F**
This PR contains the following modifications:
- csr_accessor class moved from dal::detail to dal:: namespace
- C++example for csr_accessor was added

### CODE REVIEW

File: cpp/oneapi/dal/table/csr_accessor.hpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/table/csr_accessor.hpp:2]:
```suggestion
* Copyright 2023 Intel Corporation
```

**Review by KulikovNikita — APPROVED** [review]

**Review by inteldimitrius — APPROVED** [review]
Changes look good to me

**Pull request merged**